LamestWebserver/LamestWebserver/Core/Logger.cs

```diff
@@ -259,28 +336,55 @@ public void CrashAndBurn(string msg)
             Environment.Exit(-1);
         }
 
-        internal void Log(ELoggingLevel loggingLevel, string msg)
+        private void Log(ELoggingLevel loggingLevel, string msg)
         {
-            // TODO: Implement propper logging!
-
-            ServerHandler.LogMessage($"[{loggingLevel.ToString()} \\\\\\\\ {msg}]");
+            if (_currentOutputSource != EOutputSource.None)
+            {
+                using (_loggerMutex.Lock())
+                {
+                    _multiStreamWriter.WriteLine($"[{loggingLevel.ToString()} \\\\\\\\ {msg}]");
+                }
+            }
         }
 
         /// <summary>
         /// Closes and Flushes the Logger stream.
         /// </summary>
         public void Close()
         {
-            // TODO: Close and Flush stream!
-            DebugExcept(new NotImplementedException());
+            using (_loggerMutex.Lock())
+            {
+                _multiStreamWriter.Dispose();
+            }
         }
 
         /// <summary>
         /// Opens and initializes the steam to write to.
         /// </summary>
         protected void Open()
         {
-            DebugExcept(new NotImplementedException());
+            using (_loggerMutex.Lock())
+            {
+                if (_currentOutputSource != EOutputSource.None)
+                    _multiStreamWriter = _createMultiStreamWR();
+            }
+        }
+
+        private MultiStreamWriter _createMultiStreamWR()
```

Comment:
shouldn't this be `private __void__ CreateMultiStreamWriter()` and just set `_multiStreamWriter`? opinions?

Reply:
👍